sdks/java/io/elasticsearch-tests/elasticsearch-tests-common/src/test/java/org/apache/beam/sdk/io/elasticsearch/ElasticsearchIOTestUtils.java

```diff
@@ -165,7 +168,11 @@ static long refreshIndexAndGetCurrentNumDocs(RestClient restClient, String index
       request = new Request("GET", endPoint);
       Response response = restClient.performRequest(request);
       JsonNode searchResult = ElasticsearchIO.parseResponse(response.getEntity());
-      result = searchResult.path("hits").path("total").asLong();
+      if (backenVersion == 7) {
```

Comment:
```
if (backenVersion <= 6) {
        result = searchResult.path("hits").path("total").asLong();
      } else {
        result = searchResult.path("hits").path("total").path("value").asLong();
      }
```
to be the future proof.

Reply:
Yeah, I was thinking about removing these ifs. The aim was at the beginning was to avoid falling in a not tested case. But anyway, the getBackendVersion methods protects us from running on a non-tested version